javascript/ql/src/Security/CWE-078/SecondOrderCommandInjection.ql

```diff
@@ -0,0 +1,24 @@
+/**
+ * @name Second order command injection
+ * @description Some shell programs allow arbitrary command execution via their command line arguments.
+ *              This is a second order command injection vulnerability.
+ * @kind path-problem
+ * @problem.severity error
+ * @security-severity 7.0
+ * @precision high
+ * @id js/second-order-command-line-injection
+ * @tags correctness
+ *       security
+ *       external/cwe/cwe-078
+ *       external/cwe/cwe-088
+ */
+
+import javascript
+import DataFlow::PathGraph
+import semmle.javascript.security.dataflow.SecondOrderCommandInjectionQuery
+
+from Configuration cfg, DataFlow::PathNode source, DataFlow::PathNode sink
+where cfg.hasFlowPath(source, sink)
+select sink.getNode(), source, sink,
+  "Command line argument that allows for arbitrary command execution depends on $@.",
+  source.getNode(), source.getNode().(Source).describe()
```

Comment:
I think we should mention more specifically what kind of bypass is possible. Based on the current message, I worry that users won't really notice the "second-order" nature of this injection, and simply conclude it's a FP.

So for `git` commands, mention `--upload-pack` and for `hg` mention `--config=alias.<alias>=<command>`.